# Week 1, Checkpoint 2

Graham Swain, 08/30/22

In [1]:
import pandas as pd
import numpy as np

In [2]:
impeached_df = pd.read_csv('impeach.csv')
impeached_df.index = impeached_df['Senator']

**1)** What state did Daniel Patrick Moynihan represent?

In [3]:
print(impeached_df.loc['moynihan']['State'])

NY


Senator Moynihan represented New York.

**2)** What was the standard deviation of the conservatism score for senators in 1998? **(Hint: use the "std" command to get the standard deviation.)**

In [4]:
print(np.std(impeached_df['Conservatism']))

37.29013140228926


Standard deviation of conservatism was 37.29.

**3)** Which of Tennessee's two senators was ranked as more conservative in 1998: Thompson or Graham?

In [5]:
tn = impeached_df.loc[impeached_df['State'] == 'TN']
tn.loc[tn['Conservatism'] == tn['Conservatism'].max()].iloc[0]['Senator']

'thompson'

Thompson was the more conservative of the two.

**4)** What percentage of the vote did Bill Clinton receive in New York in the 1996 election? 

In [6]:
impeached_df.loc[impeached_df['State'] == 'NY'].iloc[0]['Clinton']

59

Clinton recieved 59% of the vote.

**5) Bonus:** Senator Kay Bailey Hutchison is not coded as "hutchison". Using any method, find the row that corresponds to Kay Bailey Hutchinson. Describe the method you used. 

In [7]:
impeached_df.loc[impeached_df['Senator'].str.contains('hut')]

Senator State  Article1  Article2  Guilt  Party  Conservatism  \
Senator                                                                    
timhutch  timhutch    AR         1         1      2      1           100   
kaybhut    kaybhut    TX         1         1      2      1            92   

          Clinton  Reelection  FirstTerm  
Senator                                   
timhutch       54        2002          1  
kaybhut        44        2000          1

In [8]:
impeached_df.loc[impeached_df.index == 'kaybhut']

Senator State  Article1  Article2  Guilt  Party  Conservatism  \
Senator                                                                  
kaybhut  kaybhut    TX         1         1      2      1            92   

         Clinton  Reelection  FirstTerm  
Senator                                  
kaybhut       44        2000          1

I cheated and originally just scanned through the CSV manually, which is possible since there are only 100 entries.

For my coded answer I just searched for Senator codes that contained 'hut'. It would've been a decent place to start to begin with, since it was likely that it would contain at least part of her last name.

In [10]:
impeached_df.loc('shelby')

ValueError: No axis named shelby for object type DataFrame